include/swift/AST/ClangModuleLoader.h

```diff
@@ -40,6 +43,47 @@ enum class ClangTypeKind {
   ObjCProtocol,
 };
 
+/// A path for serializing a declaration.
+class StableSerializationPath {
+public:
+  struct ExternalPath {
+    enum IdentifierNamespace {
+      /// A tag identifier
+      Tag,
+
+      /// An ordinary name
+      Ordinary,
+
+      /// A C++ namespace
+      Namespace
+    };
+
+    ModuleDecl *M;
+    SmallVector<std::pair<IdentifierNamespace, Identifier>, 2> Path;
+  };
+private:
+  TaggedUnion<void, const Decl *, ExternalPath> Union;
```

Comment:
Is there a specific reason why you decided to add a null state here instead of having the different functions return an `Optional<StableSerializationPath>`? This option is less verbose?

Reply:
Yeah, I found it simpler.